llvm/test/CodeGen/AIE/aie2/GlobalISel/inst-select-vst_srs.mir

```diff
@@ -481,3 +481,213 @@ body: |
+    ; CHECK-NEXT: VST_SRS_S32_S64_ag_idx_imm [[COPY2]], 32, [[COPY1]], [[COPY3]], implicit-def $srsrs_of, implicit $crsat, implicit $crrnd :: (store (<8 x s32>) into unknown-address + 32)
+    ; CHECK-NEXT: [[COPY4:%[0-9]+]]:mss = COPY [[MOV_RLC_imm10_pseudo]]
+    ; CHECK-NEXT: VST_SRS_S32_S64_ag_idx_imm [[COPY2]], 0, [[COPY]], [[COPY4]], implicit-def $srsrs_of, implicit $crsat, implicit $crrnd :: (store (<8 x s32>), align 64)
+    %0:accregbank(<8 x s64>) = COPY $bml0
+    %1:accregbank(<8 x s64>) = COPY $bml1
+    %2:ptrregbank(p0) = COPY $p0
+    %3:gprregbank(s32) = G_CONSTANT i32 1
+    %5:gprregbank(s32) = G_CONSTANT i32 0
+    %4:vregbank(<8 x s32>) = G_INTRINSIC_W_SIDE_EFFECTS intrinsic(@llvm.aie2.I256.v8.acc64.srs), %0(<8 x s64>), %5(s32), %3(s32)
+    %6:vregbank(<8 x s32>) = G_INTRINSIC_W_SIDE_EFFECTS intrinsic(@llvm.aie2.I256.v8.acc64.srs), %1(<8 x s64>), %5(s32), %3(s32)
+    %7:vregbank(<16 x s32>) = G_INTRINSIC intrinsic(@llvm.aie2.concat.I512.I256), %4(<8 x s32>), %6(<8 x s32>)
+    G_STORE %7(<16 x s32>), %2(p0) :: (store (<16 x s32>))
+...
+
+---
+name:            VST_SRS_S32_S64_VST_SRS_D32_S64_CONCAT
+alignment:       16
+legalized:       true
+regBankSelected: true
+body:             |
+  bb.1.entry:
+    liveins: $cm0, $cm1, $p0, $r0
+
+    ; CHECK-LABEL: name: VST_SRS_S32_S64_VST_SRS_D32_S64_CONCAT
+    ; CHECK: liveins: $cm0, $cm1, $p0, $r0
+    ; CHECK-NEXT: {{  $}}
+    ; CHECK-NEXT: [[COPY:%[0-9]+]]:acc512 = COPY $bml0
+    ; CHECK-NEXT: [[COPY1:%[0-9]+]]:acc512 = COPY $bml1
+    ; CHECK-NEXT: [[COPY2:%[0-9]+]]:ep = COPY $p0
+    ; CHECK-NEXT: [[COPY3:%[0-9]+]]:er = COPY $r0
+    ; CHECK-NEXT: [[MOV_RLC_imm10_pseudo:%[0-9]+]]:er = MOV_RLC_imm10_pseudo 0
+    ; CHECK-NEXT: [[COPY4:%[0-9]+]]:mss = COPY [[MOV_RLC_imm10_pseudo]]
+    ; CHECK-NEXT: VST_SRS_S32_S64_ag_idx_imm [[COPY2]], 32, [[COPY1]], [[COPY4]], implicit-def $srsrs_of, implicit $crsat, implicit $crrnd :: (store (<8 x s32>) into unknown-address + 32)
+    ; CHECK-NEXT: $crsrssign = COPY [[COPY3]]
+    ; CHECK-NEXT: [[COPY5:%[0-9]+]]:mss = COPY [[MOV_RLC_imm10_pseudo]]
+    ; CHECK-NEXT: VST_SRS_D32_S64_ag_idx_imm [[COPY2]], 0, [[COPY]], [[COPY5]], implicit-def $srsrs_of, implicit $crsat, implicit $crrnd, implicit $crsrssign :: (store (<8 x s32>), align 64)
+    ; CHECK-NEXT: $crsrssign = MOV_scalar_imm10_pseudo 0
+    %0:accregbank(<8 x s64>) = COPY $bml0
+    %1:accregbank(<8 x s64>) = COPY $bml1
+    %2:ptrregbank(p0) = COPY $p0
+    %3:gprregbank(s32) = COPY $r0
+    %5:gprregbank(s32) = G_CONSTANT i32 0
+    %7:gprregbank(s32) = G_CONSTANT i32 1
+    %4:vregbank(<8 x s32>) = G_INTRINSIC_W_SIDE_EFFECTS intrinsic(@llvm.aie2.I256.v8.acc64.srs), %0(<8 x s64>), %5(s32), %3(s32)
+    %6:vregbank(<8 x s32>) = G_INTRINSIC_W_SIDE_EFFECTS intrinsic(@llvm.aie2.I256.v8.acc64.srs), %1(<8 x s64>), %5(s32), %7(s32)
+    %8:vregbank(<16 x s32>) = G_INTRINSIC intrinsic(@llvm.aie2.concat.I512.I256), %4(<8 x s32>), %6(<8 x s32>)
```

Comment:
Could you add a test with a `set_sat` intrinsic call in between `llvm.aie2.concat` and `G_STORE`?